lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +493,1297 @@ LocaleMatch lookupMatcher(
+      u"java",     u"kali",     u"khmr",     u"knda",     u"lana",
+      u"lanatham", u"laoo",     u"latn",     u"lepc",     u"limb",
+      u"mathbold", u"mathdbl",  u"mathmono", u"mathsanb", u"mathsans",
+      u"mlym",     u"modi",     u"mong",     u"mroo",     u"mtei",
+      u"mymr",     u"mymrshan", u"mymrtlng", u"newa",     u"nkoo",
+      u"olck",     u"orya",     u"osma",     u"rohg",     u"saur",
+      u"segment",  u"shrd",     u"sind",     u"sinh",     u"sora",
+      u"sund",     u"takr",     u"talu",     u"tamldec",  u"telu",
+      u"thai",     u"tibt",     u"tirh",     u"vaii",     u"wara",
+      u"wcho"};
+  // Ref::
+  // https://github.com/unicode-org/cldr/blob/release-37/common/bcp47/collation.xml
+  // -- Minus "standard" & "search" which are not allowed as per spec:
+  // https://tc39.es/ecma402/#sec-intl-collator-internal-slots
+  static std::vector<std::u16string> coVector = {
+      u"big5han",
+      u"compat",
+      u"dict",
+      u"direct",
+      u"ducet",
+      u"emoji",
+      u"eor",
+      u"gb2312",
+      u"phonebk",
+      u"phonetic",
+      u"pinyin",
+      u"reformed",
+      u"searchjl",
+      u"stroke",
+      u"trad",
+      u"unihan",
+      u"zhuyin"};
+  static std::vector<std::u16string> caVector = {
+      u"buddhist",
+      u"chinese",
+      u"coptic",
+      u"dangi",
+      u"ethioaa",
+      u"ethiopic",
+      u"gregory",
+      u"hebrew",
+      u"indian",
+      u"islamic",
+      u"islamic-umalqura",
+      u"islamic-tbla",
+      u"islamic-civil",
+      u"islamic-rgsa",
+      u"iso8601",
+      u"japanese",
+      u"persian",
+      u"roc"};
+  auto pos = iter - keys.begin();
+  switch (pos) {
+    case 0:
+      return containsValidKeyword(nuVector, value);
+    case 1:
+      return containsValidKeyword(coVector, value);
+    case 2:
+      return containsValidKeyword(caVector, value);
+    default:
+      return false;
+  }
+};
+void setUnicodeExtensions(
+    const std::u16string &supportedExtendionKey,
+    const std::vector<std::u16string> &valueList) {
+  std::vector<std::u16string> unicodeExtensionKeywords;
+  // TODO: From BCP Parser
+  // https://github.com/facebook/hermes/blob/17d7583650e5f4b59645e0607e24c7ccb15404ee/lib/Platform/Intl/java/com/facebook/hermes/intl/LocaleObjectAndroid.java#L247
+};
+// https://tc39.es/ecma402/#sec-resolvelocale
+ResolveLocale resolveLocale(
+    const std::vector<std::u16string> &availableLocales,
+    const std::vector<std::u16string> &requestedLocales,
+    const Options &options,
+    const std::vector<std::u16string> &relevantExtensionKeys
+    //  In line with Java, haven't included LocaleData
+) {
+  //  Skip 1/2/3, as we don't need to have independant implementations for best
+  //  fit
+  auto localeMatchResult = lookupMatcher(requestedLocales, availableLocales);
+  //  5. Let result be a new Record.
+  ResolveLocale result;
+  std::u16string value;
+  std::vector<std::u16string> supportedExtensionAdditionKeys;
```

Comment:
Also is there a reason why there are no **7.** and **8.** ?

Reply:
Sorry, I hadn’t yet tested any of the functionality in resolveLocale as LocaleMatcher wasn’t returning any extensions even when there were some. I had trouble writing this, as I tried to follow the structure of https://github.com/facebook/hermes/blob/1ece7cc6e59183055295829d1ba1c364cca1a805/lib/Platform/Intl/java/com/facebook/hermes/intl/LocaleResolver.java#L20 but it got complicated as the variable naming didn’t match the spec and there weren’t many comments.

Reply:
Okay, so the problem is that Java implementation does not follow the specification precisely.
https://tc39.es/ecma402/#sec-resolvelocale 7.a and 7.b say that there have to be components that are populated by this function https://tc39.es/ecma402/#sec-unicode-extension-components. And keywords is a member of components.
Java people decided to do this work in LocaleMatcher https://github.com/facebook/hermes/blob/1ece7cc6e59183055295829d1ba1c364cca1a805/lib/Platform/Intl/java/com/facebook/hermes/intl/LocaleMatcher.java#L59 , and assigned components directly to LocaleMatchResult.extensions

Reply:
I would suggest to implement https://tc39.es/ecma402/#sec-unicode-extension-components, implement **7.**, and stick to spec as much as possible.